chainer/variable.py

```diff
@@ -925,6 +925,9 @@ def _set_chainerx_grad(self, g):
                 raise RuntimeError(
                     'Cannot set a gradient to an empty variable')
         elif arr.is_backprop_required():
+            # If g is grad-stopped, require grad on it.
+            if g is not None and not g.is_backprop_required():
+                g.require_grad()
```

Comment:
Discussed offline to create a view of `g` so that the user's gradient isn't modified. This is to keep similar semantics to the array passed as the data to `Variable.__init__`.